src/lib.rs

```diff
@@ -269,9 +257,33 @@ pub mod pallet {
 	#[pallet::storage]
 	#[pallet::getter(fn last_stake_amount_update)]
 	pub(super) type LastStakeAmountUpdate<T> = StorageValue<_, Timestamp, ValueQuery>;
-	/// Mapping of query identifiers to a report.
+	/// Mapping of reported timestamps (by query identifier) to whether they have been disputed.
+	#[pallet::storage]
+	pub(super) type ReportDisputes<T> =
+		StorageDoubleMap<_, Identity, QueryId, Blake2_128Concat, Timestamp, bool, ValueQuery>;
```

Comment:
What if two values reported have the same query id and timestamp? Can't this happen if reporters use zero nonce when calling `submit_value`?

Reply:
I believe https://github.com/tellor-io/tellor-pallet/blob/ef89ff9bce01be30a67cfed7bf08fa0a123c797a/src/lib.rs#L1064-L1068 in submit_value will prevent this. The second value submitted in the same block will fail if already reported for the query_id as both will share the same block timestamp value.

So my understanding is that if a zero nonce tx is processed first, any following tx in same block (for same query id) will be rejected and likewise the same the other way, depending on tx priority.

The same applies in the source oracle contract at https://github.com/tellor-io/tellorFlex/blob/3b3820f2111ec2813cb51455ef68cf0955c51674/contracts/TellorFlex.sol#L310-L314, except there the Report struct is keyed by query id. This PR essentially just lifts out the same structures defined in the source contracts, but makes it so that individual state changes can occur without reading a report struct with possibly thousands of items within its fields (maps/array).

Reply:
Got it. Thanks for the explanation.